### PR TITLE
Add doc for creating a CustomLink component

### PR DESCRIPTION
**Pull request by sou1est**
Added documentation for `CustomLinks` with example using new `useMatch` hook. Tried to keep it as simple as possible and using only one interface for `CustomLink` props.

### CODE REVIEW

File: docs/advanced-guides/custom-links.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Creating your own custom link
+
+Very often you need to track the current active route and apply some styling to show it. 
+With the new `useMatch` hook you can create a component and check whether a given path matches current url.
```
Comment (MeiKatz) [inline docs/advanced-guides/custom-links.md:4]: > [...] matches _the_ current url.

Comment (timdorr) [inline docs/advanced-guides/custom-links.md:4]:
```suggestion
With the new `useMatch` hook you can create a component and check whether a given path matches the current url.
```

**Comment by sou1est** [comment]
@timdorr Do I need to do something more or it's okay for now?

**Comment by timdorr** [comment]
Actually, look at this, this isn't a exactly what's intended for this page. What you've built is just `NavLink` with a wrapper div. Instead, this is intended to discuss the `component` prop and building more complex link examples.

Thanks, but I think we're going to go in a different direction with this page.

**Pull request closed without merging**